tests/schemapack_/transform/test_base.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test the base module."""
+
+
+import pytest
+from pydantic import ValidationError
+
+from metldata.schemapack_.builtin_transformations.null import NULL_TRANSFORMATION
+from metldata.schemapack_.transform.base import (
+    WorkflowDefinition,
+    WorkflowStep,
+)
+
+
+def test_workflow_definition_invalid_step_refs():
+    """Test that an invalid step reference raises an error."""
+    with pytest.raises(ValidationError):
+        WorkflowDefinition(
+            description="A workflow for testing.",
+            steps={
+                "step1": WorkflowStep(
+                    description="A dummy step.",
+                    transformation_definition=NULL_TRANSFORMATION,
+                    input=None,
+                ),
+                "step2": WorkflowStep(
+                    description="Another dummy step.",
+                    transformation_definition=NULL_TRANSFORMATION,
+                    input="non_existing_step",
+                ),
+            },
+            artifacts={
+                "step1_output": "step1",
+                "step2_output": "step2",
+            },
+        )
+
+
+def test_workflow_definition_invalid_multiple_first_steps():
+    """Test that specifing multiple steps without input raises an exeception."""
+    with pytest.raises(ValidationError):
```

Comment:
```suggestion
    with pytest.raises(ValidationError,
                       match="There should be exactly one step with input=None"):
```

Reply:
Do you think matching the error message is a stable solution? If you think it is required to distinguish this errors from other errors of that type, wouldn't it be better to use subclasses?

Reply:
I added this to the refactoring list to be picked up if we decide to continue the schemapack strategy.